File: NeuroEvolution/flappyBird/block.js
class block {

    constructor() {

        this.width = 100;
        this.speed = 5;
        this.xPos = width;
        let gapSize = 70;

        // Math.floor(Math.random() * (max - min + 1)) + min     ceil(max), floor(min)

        let topStart = Math.random() * (height - 50);
        let bottomStart = Math.floor(Math.random() * (height - ((topStart + gapSize) + 1))) + (topStart + gapSize);

        this.topStart = topStart;
        this.bottomStart = bottomStart;

    }

    show() {

        fill(0);
        rect(this.xPos, 0, this.width, this.topStart);
        rect(this.xPos, this.bottomStart, this.width, height);
    }

    update() {
        this.xPos -= this.speed;
    }

    hit(agent) {

        if (agent.yPos - (agent.radius * 0.5) < this.topStart || agent.yPos + (agent.radius * 0.5) > this.bottomStart) {
            if (agent.xPos > this.xPos && agent.xPos < this.xPos + this.width) {
                return true;
            }
        }

        return false;
    }

    offscreen() {
        if (this.xPos < -this.width) {
            return true;
        } else {
            return false;
        }
    }


}

if (typeof module !== 'undefined') {
    module.exports = block;
}

File: python_models/src_STM_model/src_pycharm/block.py

from p5 import *
import numpy as np

class Block:

    def __init__(self):

        self.width = 100
        self.speed = 4
        self.xPos = width
        gapSize = 80

        topStart = np.random.rand() * (height - gapSize)
        min = topStart + gapSize
        bottomStart = np.floor(np.random.rand() * (height - min + 1)) + min

        self.topStart = topStart
        self.bottomStart = bottomStart


    def show(self):

        fill(0)
        rect(self.xPos, 0, self.width, self.topStart)
        rect(self.xPos, self.bottomStart, self.width, height)


    def update(self):
        self.xPos = self.xPos - self.speed


    def hit(self, agent):

        if agent.yPos - (agent.radius * 0.5) < self.topStart or agent.yPos + (agent.radius * 0.5) > self.bottomStart:
            if (agent.xPos > self.xPos) and agent.xPos < self.xPos + self.width:

                agent.yPos = height/2
                return True

        return False


    def offscreen(self):
        if self.xPos < -self.width:
            return True
        else:
            return False


if __name__ == "__main__":
    pass




